schematics/utils/devkit-utils/ast-utils.ts

```diff
@@ -0,0 +1,448 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+import { Change, InsertChange } from './change';
+import { insertImport } from './route-utils';
+
+
+/**
+ * Find all nodes from the AST in the subtree of node of SyntaxKind kind.
+ * @param node
+ * @param kind
+ * @param max The maximum number of items to return.
+ * @return all nodes of kind, or [] if none is found
+ */
+export function findNodes(node: ts.Node, kind: ts.SyntaxKind, max = Infinity): ts.Node[] {
```

Comment:
@brocco @hansl will these functions (and other utils in the PR) be a part of the public schematics API? It's a pretty large amount of code that schematic creators are needing to copy now. Is this just a temporary thing we can get rid of eventually, or do we need to own this code forever?

@amcdnl is there anything that you needed to customize in these utils?

Reply:
Nope, these are straight copies from that repo.

Reply:
I chatted w/ Hans and these utils are eventually going to be a standalone thing, so in the meantime we can just treat them as an external library that happens to live in our source.